scripts/EM_plot.py

```diff
@@ -0,0 +1,256 @@
+    elif b_fs * gamma_fs > 0.03 and b_fs * gamma_fs <= 1:
+        t = (10**(0.975 + (1.735 * ((b_fs / 0.1)**0.5)) + ((0.26 - 0.08 * ((b_fs / 0.1)**0.5)) * np.log10(n / 1e15)))) * 1e4
+    elif b_fs * gamma_fs > 1:
+        t = (gamma_sf_f * 5e4) / k
+    if b_fs >= 1:
+        print("Velocity is greater than speed of light, using beta = 0.98 instead")
+        b_fs = 0.98
+        gamma_fs = 1 / np.sqrt(1 - b_fs**2)
+        gamma_sf = gamma_fs / np.sqrt(2)
+        gamma_sf_f = gamma_sf**(1 + np.sqrt(3))
+        t = (gamma_sf_f * 5e4) / k
+    return t
+
+# Plancks law for frequency
+t = temp_fxn(v_fs, dens, m_p, c, k)  # Calculate temperature based on forward shock velocity and density
+
+def planck_fxn(filter, t):
+    """Planck's law for frequency
+    This function calculates the spectral radiance of a black body at a given temperature T for a specific frequency filter.
+    
+    
+    Parameters
+    ----------
+    filter (array-like): The frequency values for which to calculate the spectral radiance.
+    T (float): The temperature of the black body in Kelvin. Taken from the if functions depending on velocity of forward shock.
+    Returns
+    -------
+    B : array-like
+        Spectral radiance of the black body at the given frequency and temperature."""
+    B = (((2*h*(filter**3))/(c**2)))*((1/(np.exp((h*filter)/(k*t))-1)))
+    return B
+
+planck_r = planck_fxn(r_freq,t)
+planck_g = planck_fxn(g_freq,t)
+planck_i = planck_fxn(i_freq,t) 
+planck_u = planck_fxn(u_freq,t)
+planck_y = planck_fxn(y_freq,t)
+planck_z = planck_fxn(z_freq,t)
+
+# Combining the filter data and planck law 
+
+r_planck = planck_r * r_eff #total output for given filter in planck law
+g_planck = planck_g * g_eff
+i_planck = planck_i * i_eff
+u_planck = planck_u * u_eff
+y_planck = planck_y * y_eff
+z_planck = planck_z * z_eff
+
+r_total = sum(r_planck) #combinging the 1600ish values in each filter for total output
+g_total = sum(g_planck)
+i_total = sum(i_planck)
+u_total = sum(u_planck)
+y_total = sum(y_planck)
+z_total = sum(z_planck)
+
+#ratio of highest output to other filters
+r_ratio = r_total / i_total
+g_ratio = g_total / i_total
+i_ratio = i_total / i_total
+u_ratio = u_total / i_total
+y_ratio = y_total / i_total
+z_ratio = z_total / i_total
+
+#Planck law graph
+total_wave = np.concatenate((r_wave, g_wave, i_wave, u_wave, y_wave, z_wave)) #combining all the wavelengths in one 
+planck_total = np.concatenate((planck_r, planck_g, planck_i, planck_u, planck_y, planck_z)) #Full Planck law
+trans_total = np.concatenate((r_planck, g_planck, i_planck, u_planck, y_planck, z_planck))
+total_freq = np.concatenate((r_freq, g_freq, i_freq, u_freq, y_freq, z_freq))#sorting the frequency values
+sorted_freq = np.sort(total_freq)
+
+# To make plancks law graph smooth 
+total_index = np.argsort(total_freq)
+total_wave = total_wave[total_index]
+planck_total = (planck_total[total_index])  
+
+# planck law based on area
+planck_area = (planck_total * pi * (rad*2))
```

Comment:
Should this be `rad**2`? I'm interpreting this equation as `P_a = P_t * (2 pi R)`. But `2 pi R` is circumference not area. However, this may not be important because `planck_area` is only used to calculate `planck_disk` which is not used anywhere else. Are these parameters planned to be used in the future by some more additions to the script?